### DISCUSSION TITLE
"Done" action for modal stackPresentation in iOS

### DISCUSSION THREAD

**Discussion by mcchrish** [software-mansion/react-native-screens#929] [Ideas]
It's been requested https://github.com/software-mansion/react-native-screens/issues/800 that modal stackPresentation to have headers in iOS and it's now merged in released https://github.com/software-mansion/react-native-screens/pull/821.

The next improvement perhaps is to have a "Done" button at the right side of the header which is pretty common in iOS modal. The App Store for example:
<img width="320" alt="Screenshot 2021-05-11 at 4 09 50 PM" src="https://user-images.githubusercontent.com/7200153/117782299-10a2fe80-b274-11eb-9c3c-b848fe067dcd.png">
<img width="320" alt="Screenshot 2021-05-11 at 4 10 16 PM" src="https://user-images.githubusercontent.com/7200153/117782305-11d42b80-b274-11eb-9d32-635f81ca159e.png">
This is a good way to compensate the lack of back button when using modal.

**Comment by kacperkapusciak** [comment]
Hi @mcchrish,
you can simply add it yourself with a `headerRight` prop.
```js
headerRight: () => <Button title="Done" onPress={() => {}} />,
```
Cheers

**Reply by mcchrish** [reply]
That is currently how I've done it, but my suggestion is to have it implemented by the library itself. I think it is inline with the project's goal to be as close to how a native app would look.
